### ISSUE TITLE
segmentation fault on quick example

### ISSUE THREAD

**Issue by benijohn** [patrick-kidger/equinox#1031]
Hi, I am trying to build a simple MLP for use as a tire model in a larger dynamics model. I have built this model using PyTorch and it works well, but most of my work is in JAX so I want the model in JAX as well.  I keep getting a segmentation fault even using the simple quick example in the documentation, this one here:

import equinox as eqx
import jax

class Linear(eqx.Module):
    weight: jax.Array
    bias: jax.Array

    def __init__(self, in_size, out_size, key):
        wkey, bkey = jax.random.split(key)
        self.weight = jax.random.normal(wkey, (out_size, in_size))
        self.bias = jax.random.normal(bkey, (out_size,))

    def __call__(self, x):
        return self.weight @ x + self.bias
    

@jax.jit
@jax.grad
def loss_fn(model, x, y):
    pred_y = jax.vmap(model)(x)
    return jax.numpy.mean((y - pred_y) ** 2)

batch_size, in_size, out_size = 32, 2, 3
model = Linear(in_size, out_size, key=jax.random.PRNGKey(0))
x = jax.numpy.zeros((batch_size, in_size))
y = jax.numpy.zeros((batch_size, out_size))
grads = loss_fn(model, x, y)

It fails on the call to loss_fn(model, x, y). This is the same place in my model that it is failing as well. I am not sure where to go from here, I have verified that jax.grad() works on  simple function on its own:

# Simple test function: f(x) = x^2
def simple_function(x):
    return x ** 2

# Compute gradient
grad_fn = jax.grad(simple_function)

# Test input
x_val = 3.0

# Run forward and gradient
y_val = simple_function(x_val)
grad_val = grad_fn(x_val)

print(f"f({x_val}) = {y_val}")
print(f"f'({x_val}) = {grad_val}")

This runs no problem. I am really at a loss. Any help/thoughts is greatly appreciated.

I am using:

Ubuntu 20
CUDA 12.4
Python 3.10 
JAX 0.6.0
Equinox 0.12.2

Thank you!
